cou/cli.py

```diff
@@ -134,20 +135,25 @@ def normal(text: str) -> str:
     )
 
 
-async def apply_plan(upgrade_plan: UpgradeStep) -> None:
+async def apply_plan(upgrade_plan: UpgradeStep, non_interactive: bool) -> None:
     """Apply the plan for upgrade.
 
     :param upgrade_plan: Plan to be executed on steps.
     :type upgrade_plan: UpgradeStep
     """
+    if non_interactive:
+        await upgrade_plan.run()
+        for sub_step in upgrade_plan.sub_steps:
+            await sub_step.run()
```

Comment:
Is this ok? We are not running any `sub_sub_step`. What I mean is that we are not doing
```python
for sub_sub_step in upgrade_plan.sub_step.sub_steps:
            await sub_sub_step.run()
```

Reply:
Yes we should make this also recursive.